src/Sluggable/Mapping/Driver/Annotation.php

```diff
@@ -86,91 +86,114 @@ public function readExtendedMetadata($meta, array &$config)
-            }
-            // process slug handlers
-            $handlers = [];
-            if (is_array($slug->handlers) && $slug->handlers) {
-                foreach ($slug->handlers as $handler) {
-                    if (!$handler instanceof SlugHandler) {
-                        throw new InvalidMappingException("SlugHandler: {$handler} should be instance of SlugHandler annotation in entity - {$meta->getName()}");
-                    }
-                    if (!strlen($handler->class)) {
-                        throw new InvalidMappingException("SlugHandler class: {$handler->class} should be a valid class name in entity - {$meta->getName()}");
-                    }
-                    $class = $handler->class;
-                    $handlers[$class] = [];
-                    foreach ((array) $handler->options as $option) {
-                        if (!$option instanceof SlugHandlerOption) {
-                            throw new InvalidMappingException("SlugHandlerOption: {$option} should be instance of SlugHandlerOption annotation in entity - {$meta->getName()}");
-                        }
-                        if (!strlen($option->name)) {
-                            throw new InvalidMappingException("SlugHandlerOption name: {$option->name} should be valid name in entity - {$meta->getName()}");
-                        }
-                        $handlers[$class][$option->name] = $option->value;
-                    }
-                    $class::validate($handlers[$class], $meta);
-                }
+        if (!is_array($slug->handlers) || [] === $slug->handlers) {
+            return [];
+        }
+
+        $handlers = [];
+
+        foreach ($slug->handlers as $handler) {
+            if (!$handler instanceof SlugHandler) {
+                throw new InvalidMappingException("SlugHandler: {$handler} should be instance of SlugHandler annotation in entity - {$meta->getName()}");
             }
-            // process slug fields
-            if (empty($slug->fields) || !is_array($slug->fields)) {
-                throw new InvalidMappingException("Slug must contain at least one field for slug generation in class - {$meta->getName()}");
+            if (!strlen($handler->class)) {
+                throw new InvalidMappingException("SlugHandler class: {$handler->class} should be a valid class name in entity - {$meta->getName()}");
             }
-            foreach ($slug->fields as $slugField) {
-                $slugFieldWithPrefix = null !== $fieldNamePrefix ? ($fieldNamePrefix.'.'.$slugField) : $slugField;
-                if (!$meta->hasField($slugFieldWithPrefix)) {
-                    throw new InvalidMappingException("Unable to find slug [{$slugFieldWithPrefix}] as mapped property in entity - {$meta->getName()}");
+            $class = $handler->class;
+            $handlers[$class] = [];
+            foreach ((array) $handler->options as $option) {
```

Comment:
Do we have a way to avoid the type casting?

Reply:
since we there is the `NamedArgumentConstructor` annotation, I think there is no need for the type casting anymore.